routing/router.go

```diff
@@ -1700,20 +1690,45 @@ func (r *ChannelRouter) preparePayment(payment *LightningPayment) (
 		return nil, err
 	}
 
+	log.Tracef("Dispatching SendPayment for lightning payment: %v",
+		newLogClosure(func() string {
+			// Make a copy of the payment with a nilled Curve
+			// before spewing.
+			var routeHints [][]zpay32.HopHint
+			for _, routeHint := range payment.RouteHints {
+				var hopHints []zpay32.HopHint
+				for _, hopHint := range routeHint {
+					h := hopHint.Copy()
+					h.NodeID.Curve = nil
+					hopHints = append(hopHints, h)
+				}
+				routeHints = append(routeHints, hopHints)
+			}
+			p := *payment
+			p.RouteHints = routeHints
+			return spew.Sdump(p)
+		}),
+	)
+
 	return paySession, nil
 }
 
 // SendToRoute attempts to send a payment with the given hash through the
 // provided route. This function is blocking and will return the obtained
 // preimage if the payment is successful or the full error in case of a failure.
-func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, route *route.Route) (
+func (r *ChannelRouter) SendToRoute(hash lntypes.Hash, rt *route.Route) (
```

Comment:
Is `SendToRoute` already fit to do mpp sending with an external payment loop? Just keep calling `SendToRoute` while checking `TrackPayment` to see how the payment comes together?

Reply:
It should work as of this PR, yes